docs/release-guide.md

```diff
@@ -12,20 +12,59 @@ please list the main repository for those in this section:
 
 * Rakudo https://github.com/rakudo/rakudo
 
-## Preparation
+## Terminology
+
+The following terms are used in this document and in other related papers:
+
+- _language revision_ or _revision_
+
+  A set of specifications defining a Perl 6 language variant. Revisions are represented by symbolic names consisting of
+  a single letter of Latin alphabet and may have an alias. For example, revision `c` is `Christmas`; `d` is `Divali`.
+
+- _language version_ or _version_
+
+  Version is a string which represents full revision identification at implementation level. It consist of `6.` prefix
+  followed by revision letter. For example, `6.d`. In the language code and sometimes at other location the version
+  string might be prefixed with `v` letter to be unambiguously identified: `v6.d`
 
-Release preparation should commence well in advance of planned release date, as there's often
-a large amount of work to perform.
+- `review`
 
-### Prep Repo
+   The process of preparing a new language release.
 
-To keep various communications for the particular release in one place, and separate from general
-spec communications, a [separate repository](https://github.com/perl6/6.d-prep) is created:
+- `PREVIEW`
 
-    https://github.com/perl6/6.d-prep
+   A modifier to mark language revision which is currently is in _review_ status: `e.PREVIEW`, `6.e.PREVIEW`,
+   `v6.e.PREVIEW`.
 
-Prior to release, ensure all the planned TODO items and Issues in that repository have
-been addressed.
+## Release Status
+
+_This section contains excerpts from [this discussion](https://github.com/perl6/problem-solving/issues/31)._
+
+Language release process is based on the fact that changes are only allowed to a revision in _review_ status. Once a
+revision is released it becomes _immutable_. From this moment on the only changes allowed are fixes. From the point of
+view of a versioning system like git would look like the following diagram:
+
+```
+             6.d tag
```

Comment:
I dislike these tags because they are historical nice-to-haves but functionally there's no reason to have them. For example, a user or a dev might accidentally check out 6.d tag instead of 6.d-errata. Maybe that's not a big deal but I'd like to see a justification for having these tags.